packages/tools/devtools/devtools-view/src/components/graphs/DynamicComposedChart.tsx

```diff
@@ -56,13 +56,16 @@ interface DataPoint {
 const mergeDataSets = (dataSets: GraphDataSet[]): DataPoint[] => {
 	const xAxisDataPointToYAxisDataPointMap: Record<
 		string,
-		Record<string, number | string>
+		Record<string, number | string | undefined>
 	> = {};
 
 	for (const dataSet of dataSets) {
 		const { yAxisDataKey, xAxisDataKey, uuid } = dataSet.schema;
 		for (const dataPoint of dataSet.data) {
 			const xAxisDataPoint = dataPoint[xAxisDataKey];
+			if (xAxisDataPoint === undefined) {
+				continue;
```

Comment:
Maybe throw an error instead? I think the way we build all these objects this probably won't happen, but _if_ it does because of a bug, it's probably preferable to just throw here.